### PR TITLE
chore(dependencies): switch from es6-shim to core-js

### PR DESCRIPTION
**Pull request by marclaval**
Core-js is known to be much faster than es6-shim, especially in IEs.
Hopefully, this will make our CI faster.

### CODE REVIEW

File: modules/@angular/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "repository": <%= JSON.stringify(packageJson.repository) %>,
   "devDependencies": <%= JSON.stringify(packageJson.defaultDevDependencies) %>,
   "peerDependencies": {
-    "es6-shim": "<%= packageJson.dependencies['es6-shim'] %>",
+    "core-js": "<%= packageJson.dependencies['core-js'] %>",
     "reflect-metadata": "<%= packageJson.dependencies['reflect-metadata'] %>",
```
Comment (kylecordes) [inline modules/@angular/package.json]: Core-JS includes a reflect implementation, and Angular demonstrably works with that implementation, without the separate reflect-metadata. That may be irrelevant and there may be some reason why the standalone reflect-metadata is necessary in edge cases I haven't hit, but this appears to be an opportunity to drop an unnecessary dependency - always a good thing.

File: modules/@angular/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   "repository": <%= JSON.stringify(packageJson.repository) %>,
```
Comment (IgorMinar) [inline modules/@angular/package.json]: this package.json is obsolete and should be deleted. we no longer force polyfills onto users via peerDependencies

**Comment by IgorMinar** [comment]
the rest looks fine.

// @jasonchoimtt this will affect your PR

File: modules/@angular/package.json (hunk 2 above)
Comment (mlaval) [inline modules/@angular/package.json]: Deleted

**Comment by imcotton** [comment]
Why not drop `reflect-metadata` as well since it contained in `core-js` too? (e.g. angular/material2#505)

ping: angular/angular.io#1748

**Pull request merged**